blockdevice/util.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package blockdevice
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"mayadata.io/cstorpoolauto/unstruct"
+)
+
+const (
+	// Kind represents Kind of Kubernetes custom resource block device
+	Kind = "BlockDevice"
+)
+
+// GetCapacity returns capacity of a block device in resource.Quantity format
+// If value if not found or for an invalid capacity then it returns an error
+func GetCapacity(obj unstructured.Unstructured) (resource.Quantity, error) {
+	if obj.GetKind() != Kind {
```

Comment:
obj as well obj.Object might be nil. So nil check is required.

Reply:
This is handled here -
https://github.com/kubernetes/kubernetes/blob/4bb8df4464f44090ec1a595fbcaa60288e1e0d82/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L104

I will not add a nil check. I definitely will be adding a UT for it.